examples/scann.py

```diff
@@ -0,0 +1,261 @@
+"""
+# Faster retrieval with Scalable Nearest Neighbours (ScANN)
+
+Retrieval models are designed to quickly identify a small set of highly relevant
+candidates from vast pools of data, often comprising millions or even hundreds
+of millions of items. To effectively respond to the user's context and behavior
+in real time, these models must perform this task in just milliseconds.
+
+Approximate nearest neighbor (ANN) search is the key technology that enables
+this level of efficiency. In this tutorial, we'll demonstrate how to leverage
+ScANN—a cutting-edge nearest neighbor retrieval library—to effortlessly scale
+retrieval for millions of items.
+
+[ScANN](https://research.google/blog/announcing-scann-efficient-vector-similarity-search/),
+developed by Google Research, is a high-performance library designed for
+dense vector similarity search at scale. It efficiently indexes a database of
+candidate embeddings, enabling rapid search during inference. By leveraging
+advanced vector compression techniques and finely tuned algorithms, ScaNN
+strikes an optimal balance between speed and accuracy. As a result, it can
+significantly outperform brute-force search methods, delivering fast retrieval
+with minimal loss in accuracy.
+
+We will start with copying over code from the
+[basic retrieval example](https://github.com/keras-team/keras-rs/blob/main/examples/basic_retrieval.py).
+Data processing, model building, and training remain exactly the same. Feel free
+to skip this part if you have gone over the basic retrieval example before.
+
+Note: ScANN does not have its own separate layer in KerasRS because the ScANN
+library is TensorFlow-only. Here, in this example, we directly use the ScANN
+library and demonstrate its usage with KerasRS.
+
+## Imports
+
+Let's install the `scann` library and import all necessary packages.
+"""
+
+# ruff: noqa: E402
```

Comment:
Do you need this? The imports look like they're at the top.

Reply:
For some reason, the `"""shell` block isn't considered a string by ruff, and it keeps throwing E402.